R/model.R

```diff
@@ -221,8 +229,26 @@ resolve_early_stop_monitor <- function(early_stopping_monitor, valid_split) {
 train_batch <- function(network, optimizer, batch, config) {
   # forward pass
   output <- network(batch$x, batch$x_na_mask)
-  loss <- config$loss_fn(output[[1]], batch$y)
-
+  # if target is_multi_outcome, loss has to be applied to each label-group
+  if (max(batch$output_dim$shape) > 1) {
+    # TODO maybe torch_stack here would help loss$backward and better to shift right torch_sum at the end ?
+    outcome_nlevels <- as.numeric(batch$output_dim)
```

Comment:
It seems that `batch$output_dim` is placed on the model device, thus `as.numeric()` doesn't work if the model is on CUDA. 
```suggestion
    outcome_nlevels <- as.numeric(batch$output_dim$to(device="cpu"))
```